components/movementsensor/adxl345/adxl345.go

```diff
@@ -232,6 +331,123 @@ func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error
 	return handle.WriteByteData(ctx, register, value)
 }
 
+func getByteFromInterrupts(interrupts []string) byte {
+	var register byte
+	for _, interrupt := range interrupts {
+		register += interruptBitPosition[interrupt]
+	}
+	return register
+}
+
+func (adxl *adxl345) enableInterrupts(ctx context.Context) error {
+	register := getByteFromInterrupts(adxl.interruptsEnabled)
+	return adxl.writeByte(ctx, IntEnable, register)
+}
+
+func (adxl *adxl345) setRelevantRegisters(ctx context.Context) error {
+	if len(adxl.interruptsEnabled) == 0 {
+		return nil
+	}
+	relevantRegisters := adxl.getRelevantRegisterValues()
+	for key := range defaultRegisterValues {
+		val, ok := relevantRegisters[key]
+		if ok {
+			err := adxl.writeByte(ctx, key, val)
+			if err != nil {
+				return errors.Errorf("Unable to write byte at register %x", key)
+			}
+		} else {
+			err := adxl.writeByte(ctx, key, defaultRegisterValues[key])
+			if err != nil {
+				return errors.Errorf("Unable to write byte at register %x", key)
```

Comment:
Suppose we write half the registers and then return an error. Could we get into some bad state if the other half of them aren't written? Do we need to impose an ordering on the things we write to (which could be random, since I don't think maps have an ordering on their contents)?

Reply:
We do not need to impose any ordering - a bad state I could think of would be a situation where taps are enabled but the additional configuration registers for taps have not been set. I am not sure what would be the best way share that information as it is all configs. What do you think?